app/assets/javascripts/pages/agent-edit-page.js.coffee

```diff
@@ -108,10 +108,12 @@ class @AgentEditPage
     $(".control-link-region").show()
 
   hideEventCreation: ->
-    $(".event-related-region").hide()
```

Comment:
I think the best way to go is your third suggestion, but I feel like there's a cleaner way to use transactions for unit tests and truncation for feature specs, which should remove the need for two Travis runs. I'm going to dig around a bit.